hazelcast/include/hazelcast/client/config/SocketOptions.h

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (c) 2008-2018, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef HAZELCAST_CLIENT_CONFIG_SOCKETOPTIONS_H_
+#define HAZELCAST_CLIENT_CONFIG_SOCKETOPTIONS_H_
+
+#include "hazelcast/util/HazelcastDll.h"
+
+#if  defined(WIN32) || defined(_WIN32) || defined(WIN64) || defined(_WIN64)
+#pragma warning(push)
+#pragma warning(disable: 4251) //for dll export	
+#endif
+
+namespace hazelcast {
+    namespace client {
+        namespace config {
+            /**
+             * TCP Socket options
+             */
+            class HAZELCAST_API SocketOptions {
+            public:
+                /**
+                 * constant for kilobyte
```

Comment:
I think we dont need this comment, it doesnt say more than the variable name

Reply:
These are from Java client documentation. If needed we need to change both. I think it can stay.

Reply:
I think they are not even used in JAVA :) I sent a PR for that `https://github.com/hazelcast/hazelcast/pull/14147/files`

Reply:
As they are part of API we cant change them I guess. If there isnt any check like checkstyle for CPP we shouldnt have them here. They are probably there in Java because of checkstyle(otherwise the build fails).